h_malloc.c

```diff
@@ -1874,6 +1874,94 @@ EXPORT size_t h_malloc_object_size_fast(const void *p) {
     return SIZE_MAX;
 }
 
+inline void *h_memcpy_real(void *dst, const void *src, size_t len) {
```

Comment:
missing `static`

the parameters should be `restrict` pointers (that's the difference to `memmove(3)`)

also maybe for performance reasons use `__builtin_memcpy()` or copy the implementation from musl? (also affects `memset()` and `memmove()`)

Reply:
it doesn't compile with static

I added restrict

I changed to the __builtin variants, but they appear to be identical in my case

Reply:
Just use `__builtin_memcpy` instead or bringing parts of musl.

Reply:
that just ends up pointing to the _wrapped functions and recurses

Reply:
OK, I tried dlsym again and it appears to not crash like when I tested the first time

Reply:
this malloc actually implemented their own dlsym because additional function replacement causes even further conflicts https://github.com/GJDuck/libredfat/blob/master/redfat_memops.c#L196